### PR TITLE
portalUr > portalUrl

### PR DESCRIPTION
**Pull request by jgravois**
progress toward utilizing OAuth2 for authentication exclusively (both SAML cases described in #115 and more generically)

this change makes it possible to configure the app to use OAuth2 to sign into a custom portal _without_ being forced to refresh the page to see content.

```json
    "appId": "6IrLxMbvXIVlSWIL",
    "portalUrl": "https://server.com/gis/"
```

to do:

- [ ] do more testing to make sure i didn't break anything
- [ ] make green sign in dialog more generic and remove the gray one
- [ ] remove the old `generateToken` code
- [ ] ensure that its still possible to transfer content between orgs
- [ ] investigate reading `portalUrl` and `appId` url parameters (to deploy without touching the built source)

### CODE REVIEW

**Comment by jgravois** [comment]
to do:

- [ ] do more testing to make sure i didn't break anything
- [ ] make green sign in dialog more generic and remove the gray one
- [ ] remove the old `generateToken` code
- [ ] ensure that its still possible to transfer content between orgs

File: src/js/main.js [hunk 1]
```diff
@@ -1933,9 +1949,15 @@ require([
                 oAuthPopupConfirmation: false
             }).then(function(user) {
                 // If there is no destination or the destination is not the same as ArcGIS Online
-                if (!app.portals.destinationPortal || (app.portals.destinationPortal.portalUrl !== appInfo.portalUr)) {
```
Comment (jgravois) [inline src/js/main.js]: pretty sure `appInfo.portalUr` was just a typo.

**Comment by slibby** [comment]
@jgravois I think the latest #194 overrides this work, if that's OK with you 💯 .

**Comment by jgravois** [comment]
agreed (and happy to be superceded).

i had one unique bug fix in the PR, so i've rebased and isolated that change here.

my original proposal can be found in perpetuity in [`jgravois/oauth2-hack`](https://github.com/jgravois/ago-assistant/tree/oauth2-hack)

**Comment by rwmajor2** [comment]
Fixed in PR #194

**Pull request closed without merging**